src/LazyValue.php

```diff
@@ -19,27 +19,56 @@ final class LazyValue
     /** @var callable():mixed */
     private $factory;
 
+    private bool $memoize;
+
+    private mixed $memoizedValue = null;
+
     /**
      * @param callable():mixed $factory
+     *
+     * @deprecated directly using LazyValue's constructor is deprecated. It will be private in v2. Use named constructors instead.
      */
-    public function __construct(callable $factory)
+    public function __construct(callable $factory, bool $memoize = false, bool $calledInternally = false)
     {
         $this->factory = $factory;
+        $this->memoize = $memoize;
+
+        if (!$calledInternally) {
+            trigger_deprecation('zenstruck/foundry', '1.34.0', "directly using LazyValue's constructor is deprecated. It will be private in v2. Use named constructors instead.");
+        }
+    }
+
+    public static function new(callable $factory): self
+    {
+        return new self($factory, false);
```

Comment:
shouldn't we call `self($factory, calledInternally: true)` ?

and `self($factory, memoize: true, calledInternally: true)` in `memoize()`

otherwise, we will trigger our own deprecations

Reply:
Oops, good catch!